sbncode/FlashMatch/FlashPredict_module.cc

```diff
@@ -917,18 +933,60 @@ FlashPredict::FlashMetrics FlashPredict::computeFlashMetrics(
   }
   else {
     std::string channels;
-    for(auto oph=opH_beg; oph!=opH_end; ++oph) channels += std::to_string(oph->OpChannel()) + ' ';
+    for(auto oph : ophits) channels += std::to_string(oph.OpChannel()) + ' ';
     mf::LogError("FlashPredict")
       << "Really odd that I landed here, this shouldn't had happen.\n"
       << "sum:          \t" << sum << "\n"
       << "sum_PE:       \t" << sum_PE << "\n"
       << "sum_unPE:     \t" << sum_unPE << "\n"
-      << "opHits size:  \t" << std::distance(opH_beg, opH_end) << "\n"
+      << "opHits size:  \t" << ophits.size() << "\n"
       << "channels:     \t" << channels << std::endl;
     return {};
   }
 }
 
+FlashPredict::FlashMetrics FlashPredict::getFlashMetrics(
+  const FlashPredict::SimpleFlash& sf) const
+{
+  std::vector<recob::OpHit> ophits;
```

Comment:
I think you should be able to construct  `ophits` like so:
```
std::vector<recob::OpHit> ophits(sf.opH_beg, sf.opH_end);
```

to save some redundancy.